### ISSUE TITLE
Regarding training a mask model with my own data, could you please provide guidance on the steps involved

### ISSUE THREAD

**Issue by zxb-0** [valeoai/Maskgit-pytorch#10]
Thank you for your great work，I have some questions I would like to ask you, if you don't mind.
data_folder="/datasets_local/ImageNet/"
vit_folder="./pretrained_maskgit/MaskGIT/MaskGIT_ImageNet_256.pth"
vqgan_folder="./pretrained_maskgit/VQGAN/"
writer_log="./logs/"
num_worker=16
bsize=64
# Single GPU
python main.py  --bsize ${bsize} --data-folder "${data_folder}" --vit-folder "${vit_folder}" --vqgan-folder "${vqgan_folder}" --writer-log "${writer_log}" --num_workers ${num_worker} --img-size 256 --epoch 301 --resume
# Multiple GPUs single node
torchrun --standalone --nnodes=1 --nproc_per_node=gpu main.py  --bsize ${bsize} --data-folder "${data_folder}" --vit-folder "${vit_folder}" --vqgan-folder "${vqgan_folder}" --writer-log "${writer_log}" --num_workers ${num_worker} --img-size 256 --epoch 301 --resume 
If I want to train the mask with custom data, what changes do I need to make to this code? I've already trained my own VQGAN

**Comment by llvictorll** [comment]
Hello, 

If you want to train Maskgit on your custom data, you need to provide a pytorch Dataset for your specific dataset and add it in [get_data](https://github.com/valeoai/Maskgit-pytorch/blob/b0b2b3cc11cffd0b159f22dc1c6e73a7e8b53db3/Trainer/trainer.py#L41), with a custom name (i.e. "custom_data"). Next, you just need to run the following cmd (for a single gpu):

```
data_folder="/path/to/our/new/dataset/"
data="custom_data"
vit_folder="/path/to/save/your/maskgit"
vqgan_folder="/path/to/your/pretrained/vqgan"
writer_log="./logs/"
num_worker=16
bsize=256
python main.py --bsize ${bsize} --data-folder "${data_folder}" --vit-folder "${vit_folder}" --vqgan-folder "${vqgan_folder}" --writer-log "${writer_log}" --num_workers ${num_worker} --data ${data} --img-size 256 --epoch 300
```
